### DISCUSSION TITLE
Cursor Animation!

### DISCUSSION THREAD

**Discussion by BrandonKirbyson** [BrandonKirbyson/VSCode-Animations#7] [Announcements]
### Cursor Animation
I added a clean cursor animation based off [this](https://www.reddit.com/r/vscode/comments/11e66xh/i_made_neovide_alike_cursor_effect_on_vscode/).

![](https://raw.githubusercontent.com/BrandonKirbyson/VSCode-Animations/main/static/gifs/Cursor.gif)

#### Settings
The cursor animation can be toggled of course but also has a few more specific settings that live update.
| Setting       | Description                                  | Options                          |
| ------------- | -------------------------------------------- | -------------------------------- |
| `Color`       | The color of the cursor animation            | `Hex` or `RGB` color as `string` |
| `Style`       | The type of animation to use for the cursor. | `Block`, `Line`                  |
| `TrailLength` | The length of the trail behind the cursor.   | `Number`                         |

**Comment by Seng-Jik** [comment]
But how to enable this animation?

I was checked `Animations: Cursor Animation` already, but it has no animation on my editor.

**Reply by BrandonKirbyson** [reply]
Try running `Animations: Install Animations` and let me know if that fixes it.

**Reply by Seng-Jik** [reply]
No, it not fixed.

https://github.com/BrandonKirbyson/VSCode-Animations/assets/11296891/f61f5944-b681-414d-a8ab-b1a07556eda8

**Reply by BrandonKirbyson** [reply]
Ok, maybe just try uninstalling and reinstalling the extension. Your settings are saved in settings.json so its worth a try.

If you have custom css make sure to save that somewhere else because that will be reset.

Also, you just sent audio and not a video.

**Reply by Seng-Jik** [reply]
I tried reinstall the Visual Studio Code and extension, but it still not fixed. 
That video may need to be downloaded and played using a local media player.

```json
   "vscode_custom_css.imports": [
      "file://C:\\Users\\85397\\.vscode\\extensions\\coltwillcox.synthwave-x-fluoromachine-contrast-1.0.1\\synthwave-x-fluoromachine.css",
      "file://C:\\Users\\85397\\.vscode\\extensions\\coltwillcox.synthwave-x-fluoromachine-contrast-1.0.1\\logo.css",
      "file://C:\\Users\\85397\\.vscode\\extensions\\coltwillcox.synthwave-x-fluoromachine-contrast-1.0.1\\epic-80s-transitions.css",
      "file://c:\\Users\\85397\\.vscode\\extensions\\brandonkirbyson.vscode-animations-1.0.9/dist/updateHandler.js"
    ],
    "animations.Durations": {
        "Command-Palette": 200,
        "Tabs": 200,
        "Active": 200,
        "Scrolling": 200,
        "Smooth-Mode": 200
    },
    "animations.Enabled": true,
    "animations.Custom-CSS": false,
    "animations.CursorAnimationOptions": {
        "TrailLength": 100
    },
    "animations.CursorAnimation": true,
    "editor.cursorBlinking": "solid",
    "editor.cursorStyle": "block",
    "editor.cursorSmoothCaretAnimation": "on"
```

and https://github.com/qwreey75/dotfiles/tree/master/vscode/trailCursorEffect is not work on my machine too.

**Reply by BrandonKirbyson** [reply]
Ok sorry, I downloaded the video and it worked.

First off, do you have an animations when opening a new tab, try by just going `ctrl + n`. If not, then there might be an error on my part for getting the filepath on windows.

Secondly, you want to set `TrailLength` to something like `8`, try setting it to the default `8`. I will add a more realistic maximum to that setting in the next version.

Lastly, given you have synthwave which has custom css also, that could have possible conflicts. Although, I installed it myself and it still worked so thats just a hunch. Maybe just try disabling that theme just to try.

Try these and let me know if either work. Thank you for helping fix this issue.

**Reply by Seng-Jik** [reply]
I tried to enable minimap, and cursor animation worked, see issue #12 .
Thank you!

**Comment by bigmanBass666** [comment]
I declare!!!! The "vscode animations" is the best VS Code extension!!!!!!!!!!!!!!

The "cursor animation" is really really really really good-looking!!!!!!!!!!!!!!!!!!!!!!!!!!

The setting of "animations.UseCursorColorForCursorAnimation" simply makes this extension even more powerful!!!!

I downloaded this extension one or two years ago just because of the cursor animation. However, it couldn't switch automatically at that time, so I had to write scripts by myself to change the "settings.json", which was extremely troublesome!!!!!!! But now we can switch automatically. It's truly a miracle for humans and it has reignited my urge to write code.

So, I love this extension so much that I'm crazy about it😍😍😍😍😍😍😍😍😍💥💥💥💥💥💥

However, I have a small suggestion. That is about the speed of the cursor animation. It would be better if the speed of the cursor's animation could be adjusted.

Although setting the "TrailLength" to a very small number can also make it look fast, the cursor trail is so short~~~~~

So it would be better if the cursor animation could be very fast while the cursor trail is quite long at the same time.

https://github.com/user-attachments/assets/04dace2a-729a-4b76-988e-5653629eaa54

**Reply by TobinPalmer** [reply]
holy glazy

**Comment by bigmanBass666** [comment]
The cursor animation of Neovide is also very cool.

<https://youtube.com/clip/UgkxvqapWgsY6ORBDiWos09QuG8ZaqyRFyR_?si=bnzDlUJ9DXaU55MX>

<https://www.bilibili.com/video/BV1m84y1477C>
